src/Extensions/LinkObjectExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SilverStripe\LinkField\Extensions;
+
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * This is only intended to be added to Link and LinkArea
+ * Implemented as an extension rather than base class so it doesn't create an extra base table that needs to be joined
+ */
+class LinkObjectExtension extends DataExtension
+{
+    private static array $db = [
+        'OwnerID' => 'Int',
+        'OwnerClassName' => 'Varchar',
+    ];
```

Comment:
```suggestion
    private static array $jas_one = [
        'Owner' => DataObject::class,
    ];
```
This creates the same db fields (but with the correct types, dbforeignkey and dbclassname) and gives us the ability to go `$this->Owner()`

This should be declared directly on `Link` and `LinkArea` though. No need for it here.

Reply:
Also should probably be `Parent` instead of `Owner`